### ISSUE TITLE
SNOW-938539: Debian 12 Bookworm is incompatible with snowflake connector

### ISSUE THREAD

**Issue by jfo8001** [snowflakedb/snowflake-connector-python#1767]
### Python version

python 3.11.2

### Operating system and processor architecture

Linux-6.4.16-linuxkit-aarch64-with-glibc2.36

### Installed packages

```python
oscrypto 1.3.0
openssl version
OpenSSL 3.0.11 19 Sep 2023 (Library: OpenSSL 3.0.11 19 Sep 2023)
```

### What did you do?

```python
I used FROM debian:bookworm in a docker container

...then ran the snowflake_connector_python in various projects - all projects crashed when the docker container rebuilt to the new version of the OS.
```

### What did you expect to see?

Debian 12 moved to OpenSSL 3.0.11 from 3.0.9 and everything that uses snowflake_connector_python under the hood is broken.

The last known version that worked was debian:bookworm-20230919

The root cause of this is:
https://community.snowflake.com/s/article/Python-Connector-fails-to-connect-with-LibraryNotFoundError-Error-detecting-the-version-of-libcrypto#oscrypto

But the solution is not possible because there's no oscrypto=1.3.1

The developer has a fix, the root cause appears to be handling multiple digits in the openssl version string. When you go openssl 3.0.9 to 10 or 11 the regex starts failing in the python library.

https://github.com/wbond/oscrypto/commit/d5f3437ed24257895ae1edd9e503cfb352e635a8

### Can you set logging to DEBUG and collect the logs?

```bash
import logging
import os

for logger_name in ('snowflake.connector',):
    logger = logging.getLogger(logger_name)
    logger.setLevel(logging.DEBUG)
    ch = logging.StreamHandler()
    ch.setLevel(logging.DEBUG)
    ch.setFormatter(logging.Formatter('%(asctime)s - %(threadName)s %(filename)s:%(lineno)d - %(funcName)s() - %(levelname)s - %(message)s'))
    logger.addHandler(ch)
```

**Comment by sfc-gh-dszmolka** [comment]
thank you for raising this issue ! as you mentioned, it comes from `oscrypto` and hopefully the dev over there can release the fix soon. Tracking issue: https://github.com/wbond/oscrypto/issues/78

In the meantime until new version of `oscrypto` is made available, whoever is affected can try to
* remove `oscrypto` 
* install the fix from their main branch or the specific fix comment, respectively:
1. `pip install -I git+https://github.com/wbond/oscrypto.git` (from `main`)
OR
2. `pip install git+https://github.com/wbond/oscrypto.git@d5f3437ed24257895ae1edd9e503cfb352e635a8` (the fix commit)

since `snowflake-connector-python` requires `oscrypto<2.0.0`, once the fixed version of `oscrypto` is available, it can be automatically installed by (re)installing `snowflake-connector-python`; should pull the latest.

The KB article referenced above also references this future state, when the fixed `oscrypto` (believed to be 1.3.1) will be made available.

**Comment by eugen-fried** [comment]
Hi there,
Is actually waiting for the oscrypto library upgrade is a proper solution to this issue? The repo owner says it is his "[free time project](https://github.com/wbond/oscrypto/issues/78#issuecomment-1754157695)". Snowflake is a payed product, a gold standard in the data warehouse world and it looks weird to me that you transfer the responsibility for maintaining the connector functionality to some random dude. Maybe it's time for you to fork it and take the ownership of it?

**Comment by sfc-gh-dszmolka** [comment]
hi Eugen, this is a good point and valid concerns, and I absolutely agree with you on this of depending on a free-time project - and that's what our long term approach will be; the driver dev team already looking into the possibilities.

For the immediate workaround, yes we unfortunately need to use the already-existing workarounds (install the fixed commit), and for a short term solution (expected in coming days) wait for fixed `oscrypto` release.

**Comment by sfc-gh-dszmolka** [comment]
definitely encourage you to indeed take this up with your Account Team (Snowflake Sales rep) if you're already a Snowflake customer - addressing _all_ dependencies which happens to be one-man maintained and is historically a dependency for snowflake-connector-python is definitely very much out of scope for this particular issue, which tracks the `oscrypto` story.

**Comment by emulanob** [comment]
@sfc-gh-dszmolka thank you for the fix and indeed we'll contact our Account Manager to help you guys get traction internally.

**Comment by sfc-gh-aling** [comment]
thanks for all the feedback, we're going to remove the dependency on oscrpyto and pycryptodomex in our next connector release and we will do a release for snowsql accordingly.

I have created an announcement issue: https://github.com/snowflakedb/snowflake-connector-python/issues/1781

**Comment by jfo8001** [comment]
I just want to say thanks for taking a deeper look at this - not the kind of problem you want to be solving in a time crunch.  It's really hard from python to know how supported a library really is until someone puts their hand up and says hey yah, it's just me over here.   And to the thousands of folks who put their hands up and put something in the library, thank you too.   
https://xkcd.com/2347/

**Comment by sfc-gh-aling** [comment]
we have released v3.4.0 removing oscrypto and pycryptodomex, thank you!
